File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] arr, int pivot) {

        int n = arr.length;
        int[] ans = new int[n];
        int i = 0, j = n - 1, l = 0, r = n - 1;

        for(i = 0; i < n && j > -1; i++, j--)
        {
            if(arr[i] < pivot)
                ans[l++] = arr[i];

            if(arr[j] > pivot)
                ans[r--] = arr[j];
        }
        for(i = l; i <= r; i++)
            ans[i] = pivot;
        return ans;
    }
}

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        cnt = 0
        l1, l2 = [], []
        for i in nums:
            if i == pivot:
                cnt += 1
            elif i < pivot:
                l1.append(i)
            else:
                l2.append(i)
        return l1 + list(pivot for _ in range(cnt)) +  l2